core-container/entry.sh

```diff
@@ -23,145 +23,133 @@ function echoerr() {
    echo "ERROR: $1" >&2
 }
 
-
-function set_provider_adapters() {
-    oldKubeconfig=$KUBECONFIG
-    KUBECONFIG=
-    echo "Waiting for space manager to be ready... this may take a while."
+function wait_kcp_ready() {
+    echo "Waiting for kcp to be ready... this may take a while."
     (
-        until [ "$(kubectl logs $(kubectl get pod --selector=app=kubestellar -o jsonpath='{.items[0].metadata.name}') -c space-manager | grep '***READY***')" != "" ]; do
-            sleep 10
+        until [ "$(kubectl --kubeconfig $host_kubeconfig logs $(kubectl --kubeconfig $host_kubeconfig get pod --selector=app=kubestellar -o jsonpath='{.items[0].metadata.name}') -c kcp | grep '***READY***')" != "" ]; do
+           sleep 10
         done
     )
+}
+
+function get_kcp_kubeconfig() {
+    wait_kcp_ready
```

Comment:
Why wait for that other condition when all we really need at this point is that the Secret exists and has the sought `data` entry?